mdp/test/test_Unevenly_Sampled_Covariance.py

```diff
@@ -0,0 +1,57 @@
+"""
+Tests for the UnevenlySampledCovarianceMatrix
+"""
+from mdp.utils.covariance import CovarianceMatrix,\
+    UnevenlySampledCovarianceMatrix
+from mdp import numx
+from mdp.test._tools import assert_array_almost_equal, decimal
+
+
+def test_UnevenlySampledCovarianceMatrix1():
+    # initialize the estimators
+    cov = CovarianceMatrix()
+    uncov = UnevenlySampledCovarianceMatrix()
+    # set sample distribution parameters
+    tC = [[2., 0.5], [0.5, 1]]
+    tAvg = [0, 0]
+    # generate some data
+    x = numx.random.multivariate_normal(tAvg, tC, 300000)
+    # update the estimators
+    cov.update(x)
+    uncov.update(x, numx.ones((x.shape[0]-1,)))
+    # quit estimating
+    unC, unAvg, unTlen = uncov.fix()
+    C, avg, tlen = cov.fix()
+
+    # same for uneven, random time increments
+    uncov = UnevenlySampledCovarianceMatrix()
+    inc = (numx.random.rand(x.shape[0]-1)-.5)*.2 + 1.
+    uncov.update(x, inc)
+    unC2, unAvg2, unTlen2 = uncov.fix()
+
+    assert_array_almost_equal(unC, tC, decimal-5)
+    assert_array_almost_equal(unC2, tC, decimal-5)
+
+    prec = numx.linalg.norm(tC-C)
+    unPrec = numx.linalg.norm(tC-unC)
+    unPrec2 = numx.linalg.norm(tC-unC2)
+
+    assert_array_almost_equal(prec, unPrec, decimal-3)
+    assert_array_almost_equal(prec, unPrec2, decimal-3)
```

Comment:
``-3`` seems to me like a pretty low precision. I would expect it to be at about ``-10`` or at least ``-6`` or so. Is this value kind of sharp for the test results?

Reply:
Subtracting reduces the precision here, as it is the "-exponent". In this case it was kind of hard for me to find a reasonable value. I set the value such that the test still returns a consistent result with "reasonable probability".

Reply:
Oh sorry, misread it as e-3 and thought it was the tolearance rather than the precision. In that case, -5 looks like pretty low precision, we have `decimal == 7`, right? Is this due to trapezoidal rule vs ordinary SFA?

Reply:
Please check out the new testing. It should be more clear now. In the current version we are comparing the L2 norm of the new covariance matrix estimators deviation to the deviation standard one (deviation from the signals known covariance). If they are not worse than the deviation of the standard one up to a carefully handpicked factor (1 % for non random stepsize, 20% for random stepsize), tests pass.